fail2ban/server/filter.py

```diff
@@ -844,35 +943,50 @@ def getJournalMatch(self, match): # pragma: no cover - Base class, not used
 
 import socket
 import struct
+from .utils import Utils
 
 
 class DNSUtils:
 
 	IP_CRE = re.compile("^(?:\d{1,3}\.){3}\d{1,3}$")
 
+	# todo: make configurable the expired time and max count of cache entries:
+	CACHE_dnsToIp = Utils.Cache(maxCount=1000, maxTime=60*60)
```

Comment:
I don't like decorators in such things like caching (should be so fast and so dynamic as possible).
I think, with "fast" is all right? With "dynamic" I mean - I can use it inside cached function (not only at begin check / at end cache) for example checking of cache between another code in function (very fast got out at definitely wrong parameters etc).

Your example is not so bad, but looks like the `@functools.lru_cache` with timeout loigic extended. As already said I don't like it at all.
The hashing, I mean `args_hash = hash(args) + hash(tuple(kwargs.items()))`, is nice to have if we using it really as pattern (or decorator for each function with variable arguments). It's not necessary for single key and make it very slow (because after hash match we should check all arguments again) and I believe dict will hash it internally also (much faster as you).

So at all such patterns are very nice, but imho for lazy people, that we are not. Alright?
If such decorator would be written in pure C (like some realization of lru_cache), it's a big difference, but not if that wrote in python. Too slow, too rigid ... and too lazy.